WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -28,6 +28,7 @@ public static partial class ComWrappersSupport
     {
         private readonly static ConcurrentDictionary<string, Func<IInspectable, object>> TypedObjectFactoryCache = new ConcurrentDictionary<string, Func<IInspectable, object>>();
         private readonly static ConditionalWeakTable<object, object> CCWTable = new ConditionalWeakTable<object, object>();
+        private readonly static ConcurrentDictionary<Type, MemberInfo> TypeObjectRefFieldCache = new ConcurrentDictionary<Type, MemberInfo>();

```

Comment:
is this Bartek's suggestion?

Reply:
This is indeed an area where bartek saw high costs (retrieving custom attributes).  And one of the approaches we discussed was doing caching here to see if we can improve performance.

Reply:
I still wonder if we can enhance `IWinRTObject` to avoid all of the reflection usage altogether (at least for the .NET 5.0 projection). That way we don't need to add another cache.

Reply:
I agree I think we should still look at if we can enhance IWinRTObject and eliminate it for .NET 5 which is why I am not closing the issue.  I got this change out for now as I wanted to get this PR out with some improvements as it was getting large, then do that as an additive thing when I got it working.

My initial attempt was trying to just use NativeObject from IWinRTObject as the result when a bool was set which is set in the classes which have objreferencewrapper but it broke in composable scenarios where a managed type implementing a projected type also had a NativeObject set but that wouldn't have been previously used.  I need to think about how to handle that scenario.  Was there any other enhancements you were thinking we need to make to IWinRTObject to achieve it and that may help with that issue?

Reply:
One option would be to add a new `bool` property in `IWinRTObject` named `IsComposed`. By default, it would return `false`. We'd update the codegen for composable runtime classes so for a given class `Composable`, it is implemented as follows:

```csharp
bool IWinRTObject.IsComposed => this.GetType() != typeof(Composable);
```

Since each composable class would override this, calls to it would always used the most-derived implementation. Therefore, we'd correctly be able to detect if an object is composed (and if we can unwrap it).

Reply:
That is actually a clever idea and I think will work and looks like what we do in the constructor already for another thing.  Will give it a try.